src/views/Stack/Stack.tsx

```diff
@@ -0,0 +1,325 @@
+          scene.route === oldScene.route &&
+          scene.progress.current === oldScene.progress.current &&
+          scene.progress.next === oldScene.progress.next &&
+          scene.progress.previous === oldScene.progress.previous &&
+          scene.descriptor === oldScene.descriptor
+        ) {
+          return oldScene;
+        }
+
+        return scene;
+      }),
+      progress,
+      descriptors: props.descriptors,
+    };
+  }
+
+  state: State = {
+    routes: [],
+    scenes: [],
+    progress: {},
+    layout,
+    descriptors: this.props.descriptors,
+    // Used when card's header is null and mode is float to make transition
+    // between screens with headers and those without headers smooth.
+    // This is not a great heuristic here. We don't know synchronously
+    // on mount what the header height is so we have just used the most
+    // common cases here.
+    floaingHeaderHeight: getDefaultHeaderHeight(layout),
+  };
+
+  private handleLayout = (e: LayoutChangeEvent) => {
+    const { height, width } = e.nativeEvent.layout;
+
+    if (
+      height === this.state.layout.height &&
+      width === this.state.layout.width
+    ) {
+      return;
+    }
+
+    const layout = { width, height };
+
+    this.setState({ layout });
+  };
+
+  private handleFloatingHeaderLayout = (e: LayoutChangeEvent) => {
+    const { height } = e.nativeEvent.layout;
+
+    if (height !== this.state.floaingHeaderHeight) {
+      this.setState({ floaingHeaderHeight: height });
+    }
+  };
+
+  render() {
+    const {
+      descriptors,
+      navigation,
+      routes,
+      closingRoutes,
+      onOpenRoute,
+      onCloseRoute,
+      onGoBack,
+      getPreviousRoute,
+      getGesturesEnabled,
+      renderHeader,
+      renderScene,
+      headerMode,
+      direction,
+      onTransitionStart,
+      onGestureBegin,
+      onGestureCanceled,
+      onGestureEnd,
+      transitionSpec,
+      cardStyleInterpolator,
+      headerStyleInterpolator,
+    } = this.props;
+
+    const { scenes, layout, progress, floaingHeaderHeight } = this.state;
+    const focusedRoute = navigation.state.routes[navigation.state.index];
+
+    return (
+      <React.Fragment>
+        <ScreenContainer style={styles.container} onLayout={this.handleLayout}>
+          {routes.map((route, index, self) => {
+            const focused = focusedRoute.key === route.key;
+            const current = progress[route.key];
+            const descriptor = descriptors[route.key];
+            const scene = scenes[index];
+            const next = self[index + 1]
+              ? progress[self[index + 1].key]
+              : ANIMATED_ONE;
+
+            // Display current screen and a screen beneath. On Android screen beneath is hidden on animation finished bs of RNS's issue.
+            const isScreenActive =
+              index === self.length - 1
+                ? 1
+                : Platform.OS === 'android'
+                ? cond(eq(next, 1), 0, 1)
```

Comment:
I think we may get rid of screens support for iOS now. @kmagiera promised me to fix lies breaking this implementation.

Reply:
Right, I suggest to turn off screens on iOS for the time being. Even when fixed there are still going to be some people on the old version (via expo for example). I am also confident that the fix may not be necessary, we already use screens and panning works fine with gesture handler in the non-reanimated implementation AFAIK

Reply:
@kmagiera 
How did it happen that it worked 🤔?

Reply:
> Even when fixed there are still going to be some people on the old version (via expo for example)

I think this is not a problem coz it's enabled explictly by the user by calling `useScreens`

Reply:
@brentvatne wdyt? should I remove screens usage?

Reply:
@satya164 - we can ask people to use SDK34 or higher if the issue is already fixed in the latest version of screens, as we'll release SDK34 soon and we can bump screens for it

Reply:
is it?

Reply:
if it isn't then we can just disable screens on iOS temporarily :)

Reply:
What is the issue? If @kmagiera doesn't have time maybe someone else can fix it?

Reply:
I wonder how it was possible that the  prev version of stack was working

Reply:
I'm not sure. @brentvatne did last one and you did this one XD

Though I only had this problem on Android when implementing it here https://github.com/react-navigation/reanimated-stacks/pull/3